react/react1/week3/to-do-list-3/src/components/to-do-list.jsx

```diff
@@ -0,0 +1,300 @@
+        !taskArray[currentTaskIndex].isEditModeOn;
+    }
+    setToDoList(taskArray);
+  }
+
+  function handleTaskNameChange(taskName, taskId) {
+    console.log("handleTaskNameChange-", taskName, taskId);
+    const taskArray = [...toDoList];
+    const currentTaskIndex = taskArray.indexOf(
+      taskArray.find((x) => x.id === taskId)
+    );
+    if (currentTaskIndex !== -1) {
+      taskArray[currentTaskIndex].task = taskName;
+    }
+    setToDoList(taskArray);
+  }
+
+  return (
+    <>
+      <div className="container">
+        <label>To Do Description</label>
+        <input type="text" onChange={inputTextChangeHandler}></input>
+        <label>DeadLine</label>
+        <DatePicker selected={deadlineDate} onChange={dateChangeHandler} />
+        <label className="errorMsg">{taskNameErrorMsg}</label>
+        <span className="errorMsg">{dateErrorMsg}</span>
+        <button onClick={AddClickHandler}>Add todo</button>
+        <label>{toDoList.length} Items</label>
+      </div>
+
+      <ul className="task-list-ul">
+        {toDoList.map((taskItem) => (
+          <li key={taskItem.id}>
+            <BorderBox>
+              <TaskToDo
+                todoItem={taskItem}
+                handleTaskNameChange={(taskName, id) =>
+                  handleTaskNameChange(taskName, id)
+                }
+              />
+              <input
+                type={"checkbox"}
+                onChange={() => checkBoxStateChange(taskItem.id)}
+                checked={taskItem.isCompleted}
+              ></input>
+              <button onClick={() => DeleteTask(taskItem.id)}>Delete</button>
+              <button onClick={() => editHandler(taskItem.id)}>
+                {taskItem.isEditModeOn ? "Update" : "Edit"}
+              </button>
+            </BorderBox>
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+}
+
+function TaskToDo(props) {
+  // console.log("To do Item-", props.todoItem);
+  const [toDoTaskNameTxtBox, updateToDoTaskNameTxtBox] = useState(
+    props.todoItem.task
+  );
+  const taskNameTag = props.todoItem.isCompleted ? (
+    <s>
+      <label>{props.todoItem.task}</label>{" "}
+    </s>
+  ) : props.todoItem.isEditModeOn ? (
+    <input
+      type={"text"}
+      value={toDoTaskNameTxtBox}
+      onChange={(e) => updateToDoTaskNameTxtBox(e.target.value)}
+      onBlur={() =>
+        props.handleTaskNameChange(toDoTaskNameTxtBox, props.todoItem.id)
+      }
+    ></input>
+  ) : (
+    <label>{props.todoItem.task}</label>
+  );
+
+  return (
+    <>
+      <>{taskNameTag}</>
+      <label>{FormatDate(props.todoItem.deadLine)}</label>
+    </>
+  );
+}
+function FormatDate(date) {
+  let ye = new Intl.DateTimeFormat("en", { year: "numeric" }).format(date);
+  let mo = new Intl.DateTimeFormat("en", { month: "short" }).format(date);
+  let da = new Intl.DateTimeFormat("en", { day: "2-digit" }).format(date);
+  return `${da}-${mo}-${ye}`;
+}
+export default ToDoList;
+
+/*
```

Comment:
Remember to remove commented code. I understand that you might want to look at what you have done, but please remember to remove it.